BE/src/main/java/com/example/be/repository/category/CategoryRepository.java

```diff
@@ -0,0 +1,8 @@
+package com.example.be.repository.category;
+
+import com.example.be.domain.category.Category;
+import org.springframework.data.repository.CrudRepository;
+
+public interface CategoryRepository extends CrudRepository<Category, Long> {
```

Comment:
`nit`
Spring Data JDBC에서 `CrudRepository` 는 어떻게 동작하는지 본인만의 언어로 설명해주세요.

Reply:
제가 알고있는, 학습한 부분에 대해서만 설명해보겠습니다.

1. Repository 인터페이스는 메타 인터페이스로 아무런 기능을 가지지 않고 있습니다. 이 레포지토리를 통해 타입 계층을 형성합니다.

<img width="715" alt="스크린샷 2022-04-23 오후 3 10 02" src="https://user-images.githubusercontent.com/92818747/164882455-79e6b00f-9d0d-4e35-add2-856b0020305e.png">

<br/><br/>


2. CrudRepository는 Repository를 상속하는데 여기에는 @NoRepositoryBean 어노테이션이 존재합니다. 

<img width="701" alt="스크린샷 2022-04-23 오후 4 03 34" src="https://user-images.githubusercontent.com/92818747/164884126-5bfb93f1-7349-4f5f-bea9-b5f775fca775.png">



<br/><br/>


3. NoRepositoryBean은 타입 계층 중간에 존재하는 인터페이스에 붙이게 되는데, 이 어노테이션을 붙이게 되면 Repository 인터페이스를 상속받았기 때문에(같은 타입이기 때문에) 런타임 시 빈을 생성하지 않습니다. 

<img width="618" alt="스크린샷 2022-04-23 오후 3 55 29" src="https://user-images.githubusercontent.com/92818747/164883831-b44109e0-9835-45da-aa7e-a6d56a27b966.png">

<br/><br/>

> 빈을 생성하지 않는 다는 말이 무슨 말인지 처음에 잘 이해되지 않았는데 BaelDung 글을 보고 아래와 같이 이해했습니다. 혹시 맞게 이해했을까요?

중간 계층(CrudRepositoy)의 기능은 상속 받지만 새로운 타입의 빈을 생성(기능 확장)할 수 있다. 

```java
@Repository
interface PersonRepository extends MyUtilityRepository<Person, Long> {}
```

<br/><br/><br/>

### NoRepositoryBean에 대한 설명들 

[springdocs-1](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/NoRepositoryBean.html#:~:text=Annotation%20Type%20NoRepositoryBean&text=This%20will%20typically%20be%20used,declared%20in%20that%20intermediate%20interface.) , [springdocs-2](https://docs.spring.io/spring-data/jpa/docs/1.5.0.RELEASE/reference/html/repositories.html)

`This will typically be used when providing an extended base interface for all repositories in combination with a custom repository base class to implement methods declared in that intermediate interface.`


`Note, that the intermediate repository interface is annotated with @NoRepositoryBean. Make sure you add that annotation to all repository interfaces that Spring Data should not create instances for at runtime.`

<br/>


[Baeldung](https://www.baeldung.com/spring-data-annotations)

`@NoRepositoryBean does exactly this: when we mark a child interface of org.springframework.data.repository.Repository, Spring won't create a bean out of it.`

Reply:
위의 내용을 정리해서 저만의 용어로 정리해보면 CrudRepository는 `중간 계층을 빈으로 등록하지 않고 상속을 통해 기능을 확장할 수 있게 하는 인터페이스다`  로 정의 했는데 적고 보니 이상한것 같습니다 😔

Reply:
뭐 어렵게 보실 것은 없구요. 이 어노테이션이 붙어있으면, 실제 프록시 빈으로 등록하지 않게 해주는 어노테이션이라고 해요. 그래서 중간 단계의 Repository 들을 빈으로 등록하지 않으려고 붙여져 있는 겁니다.